src/Behat/Testwork/Exception/Stringer/PHPUnitExceptionStringer.php

```diff
@@ -35,17 +35,24 @@ public function supportsException(Exception $exception)
      */
     public function stringException(Exception $exception, $verbosity)
     {
-        if (class_exists('PHPUnit\\Util\\ThrowableToStringMapper')) {
-            return trim(\PHPUnit\Util\ThrowableToStringMapper::map($exception));
-        }
-
-        if (!class_exists('PHPUnit\\Framework\\TestFailure')) {
-            return trim(\PHPUnit_Framework_TestFailure::exceptionToString($exception));
-        }
-
         // PHPUnit assertion exceptions do not include expected / observed info in their
         // messages, but expect the test listeners to format that info like the following
         // (see e.g. PHPUnit_TextUI_ResultPrinter::printDefectTrace)
-        return trim(\PHPUnit\Framework\TestFailure::exceptionToString($exception));
+
+        switch (true) {
+            case class_exists(\PHPUnit\Util\ThrowableToStringMapper::class):
+                return trim(\PHPUnit\Util\ThrowableToStringMapper::map($exception));
+
+            case class_exists(\PHPUnit\Framework\TestFailure::class):
+                return trim(\PHPUnit\Framework\TestFailure::exceptionToString($exception));
+
+            case class_exists(\PHPUnit_Framework_TestFailure::class):
+                return trim(\PHPUnit_Framework_TestFailure::exceptionToString($exception));
```

Comment:
I'm also not a fan of this, but back in the day there seemed to be a lot of pressure on maintaining BC compatibility, even if it meant supporting extremely old unsupported PHP versions. I can tell you this class wasn't meant for PHPUnit 6, but even older. 😅 

I'm fine with removing this, if others agree as well.

Reply:
I think there's a difference between runtime BC - things shouldn't start throwing exceptions or failing scenarios - and cosmetic BC - if a scenario fails, we should always render exactly the same failure message.

Basically, would the change break someone's workflow or force them to upgrade a php version / other tool?

I don't think that applies here.

Reply:
I'm sorry, I'm not really following your point here. 🤔

Reply:
Sorry for being unclear.

My point was just that I don't believe changing the content of a failure message (which is all that would happen if we removed support for the PHPUnit_Framework_XXX class names) is a BC break. 

And therefore I think they should go, to keep our own code simpler & more maintainable.

Reply:
In that case I would disagree - since we are changing existing behaviour, it's still a BC break for me. The end user might be depending on the exact string, for one.

But anyway, as said earlier I'm fine with removing this as long as there's broad agreement.